component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -755,6 +758,15 @@ private void redirectToMobileNoReqPage(HttpServletResponse response, Authenticat
                 String mobileNumberPatternViolationError = SMSOTPConstants.MOBILE_NUMBER_PATTERN_POLICY_VIOLATED;
                 String mobileNumberPattern =
                         context.getAuthenticatorProperties().get(SMSOTPConstants.MOBILE_NUMBER_REGEX);
+                if (isMobileNumberUpdateFailed(context)) {
+                    url = url + SMSOTPConstants.RETRY_PARAMS;
+                    if (context.getProperty(SMSOTPConstants.PROFILE_UPDATE_FAILURE_REASON) != null) {
+                        String failureReason = String.valueOf(
+                                context.getProperty(SMSOTPConstants.PROFILE_UPDATE_FAILURE_REASON));
+                        String urlEncodedFailureReason = URLEncoder.encode(failureReason, CHAR_SET_UTF_8);
+                        url = url + SMSOTPConstants.ERROR_MESSAGE_DETAILS +  urlEncodedFailureReason;
```

Comment:
use FrameworkUtils.appendQueryParamStringtoURL method here

Reply:
Fixed this issue